### ISSUE TITLE
toast: needs to support non-visual & keyboard users

### ISSUE THREAD

**Issue by marcysutton** [angular/material#349]
To be accessible, `material-toast` needs to be exposed to users who cannot see it appear. It must also be operable without a mouse and be discoverable with touch. There are a few things involved:
1. Toast contents can be read aloud in an [ARIA live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) (recommended setting: "polite"). Region must be bootstrapped on page load and be part of the page's render flow (meaning it cannot be hidden with `display: none`). Offscreen CSS works fine for this. 
2. The escape key should operate in a "stack" to close actions in the order they were triggered.
3. Any actions inside of the Toast need to be discoverable and operable from the keyboard or touch (Undo, etc.). 

Escape stack tool idea from @ajoslin: https://gist.github.com/ajoslin/498892d942e4fc5c552c

**Comment by naomiblack** [comment]
@marcysutton is this still an issue...? If so, can you update with status and assign to @jelbourn if more work needed.

**Comment by calebegg** [comment]
This is still an issue. Toasts are not read out loud or keyboard accessible, at least on [the demo page](https://material.angularjs.org/HEAD/#/demo/material.components.toast).

**Comment by marcysutton** [comment]
I have reassigned it to @jelbourn, sorry for the delay!

**Comment by jelbourn** [comment]
@marcysutton While not a complete solution to this, I plan on adding the following attributes to toasts:
```html
<span role="alert" aria-relevant="all" aria-atomic="true">
   <!-- toast content -->
</span>
```

This won't do anything for the keyboard experience, but should at least make the content announced. I know it's also better to have a global announcer for everything (re #3174); hopefully this is an incremental improvement towards that. 

Do you have any feedback on this approach for the short term?

**Comment by NilsAtGoogle** [comment]
I think there are still a11y issues.  I have an internal bug filed saying it's not possible to reach a toast action via accessible navigation before the toast disappears.

**Comment by Splaktar** [comment]
I posted PR https://github.com/angular/material/pull/11424 to address these issues.
